src/main/java/fr/ppptours/cafeteria/model/tags/Tag.java

```diff
@@ -5,15 +5,34 @@
  */
 public abstract class Tag {
 
+    /**
+     * The tag's id
+     */
+    private int id;
+
     /**
      * The tag name, used to identify the tag
      */
     private String tagName;
 
-    public Tag(String tagName) {
+
+    public Tag(int id, String tagName) {
+        this.id = id;
         this.tagName = tagName;
     }
 
+    public int getId() {
+        return id;
+    }
+
+    /**
+     * Sets a new tag id
+     * @param id New tag id to set
+     */
+    public void setId(int id) {
+        this.id = id;
+    }
```

Comment:
We're not supposed to change the id after its creation so I don't know if a setter makes sense

```suggestion
    /**
     * Sets a new tag id
     * @param id New tag id to set
     */
    protected void setId(int id) {
        this.id = id;
    }
```

Reply:
i agree indeed